frontend/tests/authHelpers.ts

```diff
@@ -0,0 +1,56 @@
+import { expect, type APIRequestContext, type Page } from '@playwright/test';
+import { serverBaseUrl } from './envVars';
+import { RegisterPage } from './pages/registerPage';
+import { UserDashboardPage } from './pages/userDashboardPage';
+import type { UUID } from 'crypto';
+import { executeGql } from './gqlHelpers';
+import { LoginPage } from './pages/loginPage';
+
+export async function loginAs(api: APIRequestContext, user: string, password: string): Promise<void> {
+  const loginData = {
+    emailOrUsername: user,
+    password: password,
+    preHashedPassword: false,
+  }
+  const response = await api.post(`${serverBaseUrl}/api/login`, {data: loginData});
+  expect(response.ok()).toBeTruthy();
+  await api.storageState({path: `${user}-storageState.json`});
```

Comment:
what's this for? The docs say that it returns an object but we're not using it here. Are we using it somewhere else?

Reply:
The call with `path: filename` stores the storage state in that file. It's then used in the test that called `loginAs`, e.g.:

```js
await loginAs(request, 'manager', testEnv.defaultPassword);
const managerContext = await browser.newContext({storageState: 'manager-storageState.json'});
```

The reason for this API is because it will make it slightly easier, later on, to switch the tests to creating a "logged in browser context" just once per user, and creating a fixture to that effect. (That's also why the Playwright documentation recommends doing it this way). It does have the drawback of polluting the filesystem with not-so-temporary files. But when I tried to change so that the `loginAs` helper returns a "logged in browser context" directly, rather than storing the storage state in temporary files, I found the API to be quite awkward. Does the calling test want a browser context, or simply a `page` object? Once I have a better feel for what our tests need most often, I'll be able to update the `loginAs` API accordingly.

Reply:
I think usually a page would be fine. But Tim might know better. @myieye 
I don't think we did this in .Net though so maybe a page would be fine?

Reply:
I've looked at it again, and I think you're right that we can use a page. We need an APIRequestContext, but there's one available in `page.context`. And I think that that context will automatically set cookies returned from the API. If so, then we should be able to simply `await loginAs(page, username, password)` and that page object will suddenly be logged in. I'll make that change and see if the tests still work; if so, then this will indeed be a simpler API to use.